web-client/src/primitives/bls_key_pair.rs

```diff
@@ -0,0 +1,77 @@
+use nimiq_keys::SecureGenerate;
+use nimiq_serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+
+use super::{bls_public_key::BLSPublicKey, bls_secret_key::BLSSecretKey};
+
+/// A BLS keypair
+/// It is used by validators to vote during Tendermint rounds.
+/// This is just a wrapper around our internal BLS structs
+#[wasm_bindgen]
+pub struct BLSKeyPair {
+    inner: nimiq_bls::KeyPair,
+}
+
+#[wasm_bindgen]
+impl BLSKeyPair {
+    /// Generates a new keypair from secure randomness.
+    pub fn generate() -> BLSKeyPair {
+        let key_pair = nimiq_bls::KeyPair::generate_default_csprng();
+        BLSKeyPair::from_native(key_pair)
+    }
+
+    /// Derives a keypair from an existing private key.
+    pub fn derive(private_key: &BLSSecretKey) -> BLSKeyPair {
+        let key_pair = nimiq_bls::KeyPair::from(*private_key.native_ref());
+        BLSKeyPair::from_native(key_pair)
+    }
+
+    /// Deserializes a keypair from a byte array.
+    pub fn unserialize(bytes: &[u8]) -> Result<BLSKeyPair, JsError> {
+        let key_pair = nimiq_bls::KeyPair::deserialize_from_vec(bytes)?;
+        Ok(BLSKeyPair::from_native(key_pair))
+    }
+
+    #[wasm_bindgen(constructor)]
+    pub fn new(secret_key: &BLSSecretKey, public_key: &BLSPublicKey) -> BLSKeyPair {
+        let key_pair = nimiq_bls::KeyPair {
+            secret_key: *secret_key.native_ref(),
+            public_key: *public_key.native_ref(),
+        };
+        BLSKeyPair::from_native(key_pair)
+    }
+
+    /// Serializes to a byte array.
+    pub fn serialize(&self) -> Vec<u8> {
+        let mut vec = self.inner.serialize_to_vec();
+        vec
```

Comment:
This would remove the clippy warning:
```suggestion
        self.inner.serialize_to_vec()
```